### ISSUE TITLE
request: formats

### ISSUE THREAD

**Issue by ghost** [ebolefeysot/RomcenterPlugins#3]
Hi

Can I request the missing formats like dreamcast, gba, nds, atari, ...?
You can see code of the ucon64 tool which can read all these ROMs

https://ucon64.sourceforge.io/ucon64/readme.html

https://ucon64.sourceforge.io/

**Comment by ebolefeysot** [comment]
Hi, I'm not sure all of these consoles requires a plugin. Maybe you can provide some format documentation ?
For example, afaik, gba only uses .gba format which is a copy of the rom, so the default crc calculator is enough.
Anyway, I will finish the current romcenter release before looking for new plugins.
Until then, plugins source code is open and anybody can create a new plugin.

**Comment by ghost** [comment]
> Hi, I'm not sure all of these consoles requires a plugin. Maybe you can provide some format documentation ?

I'm a complete beginner myself considering the internal format themselves (while I use ROMs for a long time).
So at this moment, I don't know more then you :(
But I try to find docs!

Is it possible to write some explanation on the romcenter wiki about how the plugins work and when they are needed, once the current plugins are ready?

My current understanding about your plugins:

- a DAT file can either contain CRC of the whole ROM file (= this code: https://github.com/ebolefeysot/RomcenterPlugins/tree/master/arcade) or CRC of internal data inside the ROM (= romcenter plugins) (in ucon64 this is called "internal checksum", mostly made by the dumping device of the cartride)
- when the 2nd is true, we have to enable the plugins

Is this correct? Or are there more cases the plugins are needed?

I was also wondering if you can explain how you detect if a DAT file needs the plugin or not: 

see screenshot of this video

![afbeelding](https://github.com/ebolefeysot/RomcenterPlugins/assets/156975202/fcb92b83-eb2e-49ad-ad1c-ed6b089f08c1)

https://www.youtube.com/watch?v=EGMZHsE0D40

**Comment by ghost** [comment]
Some docs about formats not in romcenter (regardless if they need a plugin or not):

GBA: 
I think you're correct

NDS: 
https://dsibrew.org/wiki/DSi_cartridge_header
https://imrannazar.com/The-Smallest-NDS-File
https://web.archive.org/web/20120404151617/http://www.bottledlight.com/ds/index.php/FileFormats/NDSFormat

Dreamcast:
https://multimedia.cx/eggs/understanding-the-dreamcast-gd-rom-layout/
https://github.com/SONIC3D/gdi-utils
https://github.com/SONIC3D/GDIbuilder

WonderSwan:
http://perfectkiosk.net/stsws.html#rom_header

ColecoVision:
http://www.gooddealgames.com/articles/ColecoProgramming/cv%20programming.pdf

**Comment by ebolefeysot** [comment]
Hi, plugin auto detection is done by using some samples crc of very well known games.
For example, crc 925ffa69 is super mario all star crc calculated by the plugin. So if dat contains this crc, it should specify that snes plugin is required.
Data are specified in romcenter.config:


    
    <add key="PlugInNes" value="nes.dll,40960,A1C0DA00,131072,D44B412E" />
    
    <add key="PlugInSnes" value="snes.dll,2097152,925FFA69" />
    
    <add key="PlugInA7800" value="a7800.dll,49152,5990A9F8" />
    
    <add key="PlugInFds" value="fds.dll,131000,2D9D88C0" />
    
    <add key="PlugInGb" value="gbx.dll,65536,2C27EC70" />
    
    <add key="PlugInGbc" value="gbx.dll,2097152,047BDF80" />
    
    <add key="PlugInGenesis" value="genesis.dll,524288,F9394E97" />
    
    <add key="PlugInLynx" value="lynx.dll,131072,DAF587B1" />
    
    <add key="PlugInN64" value="n64.dll,8388608,11FB579B" />
    
    <add key="PlugInPcEngine" value="pcengine.dll,524288,91CE5156" />
    
    <add key="PlugInSms" value="sms.dll,262144,B519E833" />

**Comment by ebolefeysot** [comment]
> a DAT file can either contain CRC of the whole ROM file (= this code: https://github.com/ebolefeysot/RomcenterPlugins/tree/master/arcade) or CRC of internal data inside the ROM (= romcenter plugins) (in ucon64 this is called "internal checksum", mostly made by the dumping device of the cartride)
> when the 2nd is true, we have to enable the plugins

Not exactly, romcenter plugin will calculate a crc from the pure rom part extracted from the file.

See here: https://wiki.romcenter.com/wiki/doku.php?id=plugin
